### DISCUSSION TITLE
OpenAPI generation breaks for union of annotated Pydantic models

### DISCUSSION THREAD

**Discussion by isaacharrisholt** [fastapi/fastapi#11472] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI
from pydantic import BaseModel, Field


class Model(BaseModel):
    foo: str


RequestType = Annotated[Model, Field(..., name='Foo')] | Annotated[list[Model], Field(..., name='Foos')]


app = FastAPI()

@app.post('/')
def test_route(r: RequestType):
    return 'Ok'
```

### Description

- Open either docs page (`/docs` or `/redoc`)
- `r` parameter is listed as a query parameter rather than the request body schema

### Operating System

Linux

### Operating System Details

WSL2 Ubuntu

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.12.3

**Comment by YuriiMotov** [comment]
I think the way you defined `RequestType` is incorrect.

If this is still relevant for you, please explain what you want to achieve and let's try to solve this.
I'll close this for now, but feel free to re-open when you are ready to provide info
